compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -761,10 +761,15 @@ object Types {
     }
 
     /** The set of abstract term members of this type. */
-    final def abstractTermMembers(implicit ctx: Context): Seq[SingleDenotation] = {
+    final def abstractTermMembers(implicit ctx: Context): Seq[SingleDenotation] = abstractTermMembers { m =>
+      m.is(Deferred) && ! m.matchingMember(defn.ObjectType).exists
```

Comment:
I replaced the predicate `_.is(Deferred)` with this filter, since methods of `ObjectType` are always generated automatically, so they are not really `abstract`, is it correct?

Reply:
Sorry for the delay in responding, I would prefer if we did not change the definition of `abstractTermMembers` and limited the change to `samMethod`, otherwise this might have unintended consequences, for example the following should always emit an error:
```scala
scala> class Foo { def equals(x: Any): Boolean }
1 |class Foo { def equals(x: Any): Boolean }
  |      ^
  |class Foo needs to be abstract, since def equals(x: Any): Boolean is not defined
```

Reply:
@smarter, the compiler already emits the error from the code from your example with the changed `abstractTermMembers` method.

I changed `abstractTermMembers` because in terms of processing `SAM`s the method happens to be called from three different places (links follow to the exact lines in the files):

1. [`compiler/src/dotty/tools/dotc/core/Types.scala:4372`](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/core/Types.scala#L4372)
2. [`compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala:849`](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala#L849)
3. [`tools/dotc/transform/ExpandSAMs.scala:49`](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/transform/ExpandSAMs.scala#L49)

In the whole project the method is accessed in these three places plus once in [`compiler/src/dotty/tools/dotc/typer/RefChecks.scala:483`](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/typer/RefChecks.scala#L483). I used the following command to find it out:
```bash
find . -name "*.scala" -not -path "*./.git/*" -type f -exec grep --color "abstractTermMembers" {} \+
```
So I assumed it was okay to change the method. 

I reverted the changes in the `abstractTermMembers` and added a new commit (so there are two of them in the branch now) in order to be able to get back to the previous changes if we decide it is legal to alter the method and stick to the original solution.

Reply:
> the compiler already emits the error from the code from your example with the changed abstractTermMembers method.

That's surprising since it calls abstractTermMembers, shouldn't the equals method not be returned ?

> I changed abstractTermMembers because in terms of processing SAMs the method happens to be called from three different places (links follow to the exact lines in the files):

To avoid the duplication without changing the meaning of abstractTermMembers, I suggest defining a `possibleSamMethods` that just calls abstractTermMembers and add the extra filtering (you can also add the `isSuperAccessor` filtering used in ExpandSAMs), the documentation of `possibleSamMethods` can then cite the part in https://docs.oracle.com/javase/8/docs/api/java/lang/FunctionalInterface.html which talks about the special handling of methods of java.lang.Object.

Reply:
@smarter , I extracted the common method `possibleSamMethods`, please review.
There are still two commits, I will squash them when we are ready to merge the PR.